app/routes.py

```diff
@@ -1,2 +1,253 @@
+            return make_response({"message": "Invalid data"}, 400)
+        video.title = form_data.get("title")
+        video.release_date = form_data.get("release_date")
+        video.total_inventory = form_data.get("total_inventory")
+        db.session.commit()
+        return make_response(
+            video.to_json()
+        , 200)
+    elif request.method == "DELETE":
+        db.session.delete(video)
+        db.session.commit()
+        return make_response({"id": int(video_id)}, 200)
+
+###Wave 2 Custom Endpoints
+def is_id(input):
+    condition = isinstance(input, int)
+    return True if condition else False
+
+@rental_bp.route("/check-out", methods = ["POST"])
+def check_out():
+    request_body = request.get_json()
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+
+    if not is_id(customer_id):
+        return make_response({"error": "customer ID must be an integer"}, 400)
+    elif not is_id(video_id):
+        return make_response({"error": "customer ID must be an integer"}, 400)
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    if not customer:
+        return customer_not_found(customer_id)
+    if not video:
+        return video_not_found(video_id)
+
+    if video.available_inventory == 0:
+        return make_response({"details": "No inventory available"}, 400)
+
+    video.available_inventory -= 1
+    customer.videos_checked_out_count += 1
+    due_date = datetime.now() + timedelta(days=7)
+
+    new_rental = Rental(customer_id = request_body["customer_id"], video_id = request_body["video_id"], due_date = due_date)
+    db.session.add(new_rental)
+    db.session.commit()
+    
+    check_out_rental = new_rental.to_json()
+    check_out_rental["videos_checked_out_count"] = customer.videos_checked_out_count 
+    check_out_rental["available_inventory"] = video.available_inventory
+
+    return make_response(check_out_rental, 200)
+
+@rental_bp.route("/check-in", methods = ["POST"])
+def check_in():
+    request_body = request.get_json()
+    customer_id = request_body["customer_id"]
+    video_id = request_body["video_id"]
+
+    if not is_id(customer_id):
+        return make_response("customer ID must be an integer", 400)
+    elif not is_id(video_id):
+        return make_response("video ID must be an integer", 400)
+
+    customer = Customer.query.get(customer_id)
+    video = Video.query.get(video_id)
+
+    if not customer:
+        return customer_not_found(customer_id)
+    if not video:
+        return video_not_found(video_id)
+    
+    if customer.videos_checked_out_count == 0:
+        return make_response({"details": "All videos are returned"}, 400)
+
+    video.available_inventory += 1
+    customer.videos_checked_out_count -= 1
+
+    new_rental = Rental(customer_id = request_body["customer_id"], video_id = request_body["video_id"])
+    db.session.add(new_rental)
+    db.session.commit()
```

Comment:
This creates a new entry in rental with a null due date.  It does not remove the existing rental.  The rental will still show up on /customers/<id>/rentals or /vidoes/<id>/rentals.  How are you recording that the rental has ended?